saskatoon/harvest/forms.py

```diff
@@ -495,24 +529,58 @@ def clean_about(self):
             Harvest.Status.PENDING,
             Harvest.Status.CANCELLED,
         ] and (not about or is_quill_html_empty(json.loads(about).get('html'))):
-            raise forms.ValidationError(
-                _('Please fill in the public announcement to be published on the calendar.')
-            )
+            raise forms.ValidationError(_(
+                "Please fill in the public announcement to be published on the calendar."
+            ))
 
         return about
 
-    def clean(self):
+    def clean_equipment_point(self: Self) -> Optional[Organization]:
+        """Clear equipment point if no longer available or if harvest is not scheduled or ready"""
+        equipment_point = self.cleaned_data['equipment_point']
+        if equipment_point is None:
+            return None
+
+        if self.cleaned_data['status'] not in [
+            Harvest.Status.SCHEDULED,
+            Harvest.Status.READY,
+            Harvest.Status.SUCCEEDED,
+        ]:
+            raise forms.ValidationError(_(
+                "You cannot reserve an equipment point if "
+                "the harvest is not yet scheduled or ready."
+            ))
+
+        harvest_id = self.cleaned_data['id']
+        if harvest_id is None:
+            return None
+        harvest = Harvest.objects.get(pk=harvest_id)
+
+        start = self.cleaned_data['start_date']
+        end = self.cleaned_data['end_date']
+        available_points = available_equipment_points(start, end, harvest)
+        if available_points.filter(pk=equipment_point.pk).count() != 1:
+            raise forms.ValidationError(_(
+                "The {} equipment point is no longer available."
+            ).format(equipment_point.civil_name))
+
+        return equipment_point
+
+    def clean(self: Self) -> dict[str, Any]:
         """Make sure pick_leader and status fields are compatible"""
         data = super().clean()
 
+        if data is None:
+            raise forms.ValidationError(_("The form data was empty, please try again"))
+
```

Comment:
I don't think it's worth raising a user-facing validation error just to satisfy mypy (i.e. `clean()` being seen as mypy by returning `Optional[dict[str, Any]]` )